subprojects/workers/src/main/java/org/gradle/workers/internal/WorkerDaemonServer.java

```diff
@@ -141,6 +148,15 @@ public HashCode getClassLoaderHash(@Nonnull ClassLoader classLoader) {
         DomainObjectCollectionFactory createDomainObjectCollectionFactory(InstantiatorFactory instantiatorFactory, ServiceRegistry services) {
             return new DefaultDomainObjectCollectionFactory(instantiatorFactory, services, CollectionCallbackActionDecorator.NOOP, MutationGuards.identity());
         }
+
+        IsolatedAntBuilder createIsolatedAntBuilder() {
+            DefaultModuleRegistry moduleRegistry = new DefaultModuleRegistry(CurrentGradleInstallation.get());
+            return new DefaultIsolatedAntBuilder(
+                new DefaultClassPathRegistry(new DefaultClassPathProvider(moduleRegistry)),
+                new DefaultClassLoaderFactory(),
```

Comment:
Similarly for these. It looks like maybe these could move up to WorkerSharedUserHomeScopeServices or WorkerSharedGlobalScopeServices?

I'm OK with IsolatedAntBuilder only being available at the build/worker levels separately.

Reply:
I've pulled up everything it makes sense to pull up.  The only thing I didn't pull up is `ClassPathRegistry` as this gets created with additional stuff in `GlobalScopeServices`.  We _could_ just use the same registry in the worker services (with more stuff) but then that would pull some additional services (i.e. `PluginModuleRegistry`) into worker services, which it doesn't need.  Felt better to not pull unnecessary stuff into the worker, so `ClassPathRegistry` is created differently in both places.  I don't feel strongly about this, though.